pkg/virt-controller/watch/vmi.go

```diff
@@ -332,7 +331,23 @@ func (c *VMIController) updateStatus(vmi *virtv1.VirtualMachineInstance, pods []
 		if !podExists {
 			controller.RemoveFinalizer(vmiCopy, virtv1.VirtualMachineInstanceFinalizer)
 		}
-	case vmi.IsRunning() || vmi.IsScheduled():
+	case vmi.IsRunning():
+		if !conditionManager.HasCondition(vmiCopy, virtv1.VirtualMachineInstanceReady) {
```

Comment:
`virt-handler` needs to set that condition. When the VMI is running, it is cimpletely owned by `virt-handler`.